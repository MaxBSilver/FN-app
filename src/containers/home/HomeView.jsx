import React, { Component } from "react";
import banner from "../../assets/banners/fn_background.jpg";
export class HomeView extends Component {
  render() {
    return (
      <section className="home-view">
        <div>
          <div className="home-banner-cntr">
            <img className="home-banner" src={banner} alt="banner" />
          </div>
        </div>
        <section className="home-controls-cntr">
            <div>

            </div>
            <div>
                <button>PLAY FREE NOW!</button>
            </div>
        </section>
      </section>
    );
  }
}

export default HomeView;
