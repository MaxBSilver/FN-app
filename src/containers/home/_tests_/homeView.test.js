import { mapStateToProps,  HomeView } from "../HomeView";
import React from "react";

import { shallow } from "enzyme";
describe("HomeView", () => {
  describe("HomeView Component", () => {
    let wrapper;
    beforeEach(() => {
      const mockProps = { news: [{ itemId: 1, title: "max" }] };
      wrapper = shallow(<HomeView {...mockProps} />);
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe("mapStateToProps", () => {
    it("should return an object with the news array", () => {
      const mockState = {
        news: [{ title: "New Skins" }, { title: "New Items" }]
      };
      const expected = {
        news: [{ title: "New Skins" }, { title: "New Items" }]
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
    it("should return an boolean", () => {
      const mockState = {
        isLoading: false
      };
      const expected = {
        isLoading: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
    it("should return an error message", () => {
      const mockState = {
        error: "Has Errored!"
      };
      const expected = {
        error: "Has Errored!"
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
