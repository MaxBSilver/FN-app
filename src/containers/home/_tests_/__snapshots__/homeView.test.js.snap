// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView HomeView Component should match snapshot 1`] = `ShallowWrapper {}`;
